test/models/learning_outcome_test.rb

```diff
@@ -0,0 +1,133 @@
+require 'test_helper'
+class LearningOutcomeTest < ActiveSupport::TestCase
+
+  def setup
+    data = {
+        code: 'COS10001',
+        name: 'Testing in Unit Tests',
+        description: 'Test unit',
+        # teaching_period: TeachingPeriod.find(3)
+      }
+    @unit = Unit.create(data)
+  end
+
+  # Check abbreviation uniqueness
+  def test_create_learning_outcome_with_unique_abbrev
+    #Create new unit.
+    learning_outcome_count = LearningOutcome.count
+    unit_1 = {
+        unit_id: 'ASD10001',
+        ilo_number: learning_outcome_count + 1,
+        abbreviation:'AI',
+        name:'Artificial Intelligence',
+        description:'AI'
+    }
+
+    learning_outcome = LearningOutcome.create!(unit_1)
+
+    learning_outcome_count = LearningOutcome.count
+    unit_2 = {
+        unit_id: 'ASD10002',
+        ilo_number: learning_outcome_count + 1,
+        abbreviation: 'PR',
+        name: 'Programming',
+        description: 'Create programming'
+    }
+
+    learning_outcome2 = LearningOutcome.create!(unit_2)
+    
+    #Check the abbreviation is not empty.
+    refute_empty learning_outcome.abbreviation    
+    refute_empty learning_outcome2.abbreviation
+
+    #Check the abbreviation of the unit is unique.
+    refute_equal learning_outcome, learning_outcome2
```

Comment:
This isn't checking abbreviations...

Reply:
I'll stop here...

I suggest you make a very simple test first to demonstrate that the basics of learning outcomes are tested. Create the learning outcome and then test that everything about it is setup correctly, and that you can interact with it via its relationships etc.